storage/storage.go

```diff
@@ -350,8 +351,19 @@ func structToMap(str interface{}) (map[string]interface{}, error) {
 	for i := 0; i < v.NumField(); i++ {
 		// gets us a StructField
 		fi := typ.Field(i)
-		// set key of map to value in struct field
-		out[fi.Name] = v.Field(i).Interface()
+
+		// If the field is a map we need to properly marshal it
```

Comment:
Can you be more specific in this comment? What does "properly marshal it" actually means? Why a map needs different handling than the common case? As it is now, this comment is not particularly useful.

Reply:
I rewrote the comment.